<?xml version="1.0"?>
<launch>
    <arg name="freq"              default="50"/>
    <arg name="pub_odom_topic"    default="true"/>
    <arg name="odom_topic"        default="odom"/>
    <arg name="odom_raw_topic"    default="odom_raw"/>
    <arg name="odom_frame"        default="odom"/>
    <arg name="base_frame"        default="base_footprint"/>
    <arg name="map_frame"         default="map"/>
    <arg name="cmd_vel"           default="jetauto_controller/cmd_vel"/>
    <arg name="imu_raw_topic"     default="imu_raw"/>
    <arg name="imu_topic"         default="imu"/>
    <arg name="imu_link"          default="imu_link"/>

    <!--imu过滤-->
    <include file="$(find jetauto_peripherals)/launch/imu.launch">
    <arg name="freq"          value="$(arg freq)"/>
    <arg name="imu_raw_topic" value="$(arg imu_raw_topic)"/>
    <arg name="imu_topic"     value="$(arg imu_topic)"/>
    <arg name="imu_link"      value="$(arg imu_link)"/>
    </include>

    <!--odom发布-->
    <node name="jetauto_odom_publisher" pkg="jetauto_controller" type="odom_publisher.py" required="true" output="screen">
       <rosparam file="$(find jetauto_controller)/config/calibrate_params.yaml" command="load"/>
    <param name="freq"              value="$(arg freq)"/>
    <param name="pub_odom_topic"    value="$(arg pub_odom_topic)"/>
    <param name="odom_topic"        value="$(arg odom_raw_topic)"/>
    <param name="base_frame_id"     value="$(arg base_frame)"/>
    <param name="odom_frame_id"     value="$(arg odom_frame)"/>
    <param name="cmd_vel"           value="$(arg cmd_vel)"/>
    </node>

    <!--ekf融合-->
    <include file="$(find jetauto_slam)/launch/include/ekf.launch">
        <arg name="odom_topic"      value="$(arg odom_topic)"/>
        <arg name="odom_raw_topic"  value="$(arg odom_raw_topic)"/>
        <arg name="imu_topic"       value="$(arg imu_topic)"/>
        <arg name="map_frame"       value="$(arg map_frame)"/>
        <arg name="base_frame"      value="$(arg base_frame)"/>
        <arg name="odom_frame"      value="$(arg odom_frame)"/>
    </include>
</launch>
